pkg/querier/batch/stream.go

```diff
@@ -122,10 +122,18 @@ func mergeStreams(left, right batchStream, result batchStream, size int) batchSt
 			right.next()
 		} else {
 			if (rt == chunkenc.ValHistogram || rt == chunkenc.ValFloatHistogram) && lt == chunkenc.ValFloat {
-				// Prefer histograms over floats. Take left side if both have histograms.
+				// Prefer histograms than floats. Take left side if both have histograms.
 				populate(right, rt)
 			} else {
 				populate(left, lt)
+				if hPool != nil && rt == chunkenc.ValHistogram {
```

Comment:
This might be a tiny bit cheaper depending on optimization/register use
```suggestion
				if rt == chunkenc.ValHistogram && hPool != nil {
```